handshake/src/types/lockingscript.rs

```diff
@@ -0,0 +1,176 @@
+//! Handshake LockingScript and WitnessProgram
+
+use std::io::{Read, Write};
+
+use riemann_core::{
+    ser::ByteFormat,
+    types::tx::RecipientIdentifier
+};
+
+wrap_prefixed_byte_vector!(
+    WitnessProgram
+);
+
+impl From<[u8; 20]> for WitnessProgram {
+    fn from(v: [u8; 20]) -> Self {
+        let hash = vec![0x14];
+        hash.append(&mut v.to_vec());
+        WitnessProgram::new(hash)
+    }
+}
+
+impl From<WitnessProgram> for [u8; 20] {
+    fn from(w: WitnessProgram) -> [u8; 20] {
+        let data = [0; 20];
+        data.clone_from_slice(&w.0[..]);
+        data
+    }
+}
+
+impl From<WitnessProgram> for [u8; 32] {
+    fn from(w: WitnessProgram) -> [u8; 32] {
+        let data = [0; 32];
+        data.clone_from_slice(&w.0[..]);
+        data
+    }
+}
+
+impl From<WitnessProgram> for Vec<u8> {
+    fn from(w: WitnessProgram) -> Vec<u8> {
+        w.0[1..].to_vec()
+    }
+}
+
+pub struct LockingScript {
```

Comment:
You need a derive here

`#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, Eq, PartialEq, Default)]`